include/counter/system.h

```diff
@@ -0,0 +1,121 @@
+#ifndef SYSTEM_H
+#define SYSTEM_H
+
+#include <iostream>
+#include <sstream>
+
+#if defined(__ANDROID__)
+  #include <sys/system_properties.h>
+  #include <jni.h>
+#endif
+
+#include "device_info.h"
+extern std::string uuid_;
+using namespace std;
+
+/*! \brief class System (Abstract Class) 
+ */
+class System {
+ public:
+  virtual std::string get_device_info() const =0;
+  virtual std::string get_device_id() const =0;
+  virtual ~System() {};
+};
+
+class ARM : public System {
+};
+
+class Linux_ARM: public ARM {
+ public:
+  Linux_ARM()
+  {
+    device = new DeviceInfo();
+    #if defined(__LINUX__) ||  defined(__linux__)
+    #endif
+  }
+  ~Linux_ARM()
+  {
+    delete device;
+  }
+  std::string get_device_info() const
+  {
+    return device->get_info();
+  }
+  std::string get_device_id() const
+  {
+    return device->uuid;
+  }
+ private:
+  DeviceInfo* device;
```

Comment:
It seems get_device_info and get_device_id are implemented the same for all inherited classes. Can we move those into virtual functions instead?

Reply:
We followed here Dependency inversion principle and keep System class as a Thin Interface.
These two functions depends on a Device class object. Hence, we kept implementation at respective System type derived classes.